internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -206,21 +224,47 @@ func updateFleetConfig(log *logger.Logger, src remote.Config, dst *remote.Config
 		urlCopy := *src.Transport.Proxy.URL
 		dst.Transport.Proxy.URL = &urlCopy
 
-		log.Debug("received proxy from fleet, applying it")
+	}
+
+	if src.Transport.TLS != nil {
+
+		if dst.Transport.TLS == nil {
+			dst.Transport.TLS = new(tlscommon.Config)
+		} else {
+			// copy the TLS struct
+			tlsCopy := *dst.Transport.TLS
+			dst.Transport.TLS = &tlsCopy
+		}
+
+		emptyCertificate := tlscommon.CertificateConfig{}
```

Comment:
nit: can go inside if statement to reduce lifetime of this one
`if emptyCertificate := tlscommon.CertificateConfig{}; src.Transport.TLS.Certificate == emptyCertificate {`

Reply:
Tried to do it but the go compiler complains:
```shell
internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go:240:23: syntax error: cannot use emptyCertificate := tlscommon.CertificateConfig as value
```
so I substituted in a small function (it will probably be inlined anyways)